src/seedu/addressbook/data/person/Contact.java

```diff
@@ -0,0 +1,35 @@
+package seedu.addressbook.data.person;
+
+/**
+ * Parent class of Address, Email and Phone
+ */
+public class Contact {
+	public final String value;
+	private boolean isPrivate;
+
+	public Contact(String value, boolean isPrivate) {
+		this.value = value;
+		this.isPrivate = isPrivate;
+	}
+
+	@Override
+	public String toString() {
+		return value;
+	}
+
+	@Override
+	public boolean equals(Object other) {
+		return other == this // short circuit if same object
+				|| (other instanceof Contact // instanceof handles nulls
```

Comment:
can you see why this is a problem? you can now potentially compare `Address` with `Email`, why?